.github/workflows/ci.yaml

```diff
@@ -95,16 +95,90 @@ jobs:
         run: make test
   integration_test:
     runs-on: ubuntu-latest
+    env:
+      DOCKER_CACHE: /tmp/docker-cache
     steps:
       - uses: actions/checkout@v3
+
+      - run: |
+          mkdir -p ${DOCKER_CACHE}/web
+          mkdir -p ${DOCKER_CACHE}/tests
+
+      - name: Compute cache key
+        run: echo "CACHE_KEY=$(git ls-tree --full-tree -r --name-only HEAD | grep -v -f RemoteSettings.dockerignore | git hash-object --stdin)" >> $GITHUB_ENV
```

Comment:
Probably deserves a comment -- the intent is to create a new cache key only when a file that we use in our Docker context changes. So `hash(<files checked into git> - <files we ignore in the docker container>)`